jax/_src/xla_bridge.py

```diff
@@ -881,6 +881,9 @@ def backends() -> dict[str, xla_client.Client]:
     default_priority = -1000
     for platform, priority, fail_quietly in platform_registrations:
       try:
+        if platform == "cuda" and not hardware_utils.has_visible_nvidia_gpu():
```

Comment:
a) you should do this for both cuda and rocm, and
b) this isn't an ideal place for this, because what should happen is that we *should* initialize the backend but that initialization shouldn't fail if there are no devices. i.e. you should do this check inside the plugin initialization (in xla/pjrt).

If you address (a) for now and do (b) in a follow-up PR, that would work for me.

Reply:
Ack on b). 

I don’t have an AMD card to test with so I can only guess on the implementation. I think the AMD equivalent is checking `/dev/dri/…`?

Reply:
@hawkinsp As discussed offline, we won’t make the equivalent change for AMD right now because we can’t test it.

A follow-up PR will handle this inside the plugin instead.